app/src/main/java/com/github/h3lp3rs/h3lp/signin/SignInActivity.kt

```diff
@@ -38,14 +39,24 @@ class SignInActivity : AppCompatActivity() {
     /**
      * Check if the current user is already signed in and update activity accordingly
      */
-    private fun checkIfSignedIn() {
-        if (signInClient.isSignedIn()) {
-            userUid = signInClient.getUid()
-            username = getInstance().currentUser?.displayName?.substringBefore(" ")
+    private fun offlineCheckIfSignedIn(){
+        userSignIn = storageOf(Storages.SIGN_IN) // Fetch from storage
+        if(userSignIn.getBoolOrDefault(getString(R.string.KEY_USER_SIGNED_IN), false)){
+            userUid = userSignIn.getStringOrDefault(getString(R.string.KEY_USER_UID),"")
+            username = userSignIn.getStringOrDefault(getString(R.string.KEY_USER_NAME),"")
             checkToSAndLaunchIfNotAcceptedElseMain()
         }
     }
 
+    /**
+     * Save the user authentication information to the local storage
+     */
+    private fun saveAuthentication(){
+        userSignIn.setBoolean(getString(R.string.KEY_USER_SIGNED_IN), true)
+        userUid?.let { userSignIn.setString(getString(R.string.KEY_USER_UID), it) }
+        username?.let { userSignIn.setString(getString(R.string.KEY_USER_NAME), it) }
+    }
+
```

Comment:
You could define all three as variables as they are used twice:
getString(R.string.KEY_USER_SIGNED_IN)
getString(R.string.KEY_USER_UID)
getString(R.string.KEY_USER_NAME)